### PR TITLE
Rework bundles

### PR DESCRIPTION
**Pull request by nd0ut**
From now, there is three types of bundles:

## IIFE
Full IIFE bundle with all dependencies included. It is supposed to be added via the script tag.

## CJS
Full CJS bundle without dependencies. It will contain all the namespaces: UploadAPI, RestAPI and so on. Supposed to be used with CJS-bundlers.

## ESM
Partial ESM bundles without dependencies. Separate bundles for each namespace, so we can partially import some methods, namespaces or even the whole library.

```
import * as UploadcareAPI from '@uploadcare/api'
import {UploadAPI, RestAPI} from '@uploadcare/api'
import {request} from '@uploadcare/api/upload'
import {auth} from '@uploadcare/api/rest'
```

We do not need `browser` property at the `package.json` since we do not have any environment specific code. We have `form-data` package in the dependencies, which have such code and it is have `browser` property. So this is user's bundler responsibility - how to bundle our dependencies.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,11 +2,10 @@
   "name": "@uploadcare/api",
   "version": "1.0.0-alpha",
   "description": "JavaScript library for work with Uploadcare API",
-  "main": "src/uploadcare.cjs.js",
-  "module": "dist/uploadcare.esm.js",
-  "browser": "dist/uploadcare.umd.js",
+  "main": "dist/uploadcare.cjs.js",
+  "module": "index.js",
```
Comment (Zmoki) [inline package.json]: are you sure? I think esm bundle for module and iife for browser is ok

File: rollup.config.js [hunk 2]
```diff
@@ -3,52 +3,65 @@ import babel from 'rollup-plugin-babel'
...
     ],
   },
   {
     input: 'src/index.js',
-    plugins: getPlugins({forBrowser: true}),
+    plugins: getPlugins({iife: true}),
     output: [
       {
-        file: 'dist/uploadcare.umd.js',
+        file: 'dist/uploadcare.iife.js',
```
Comment (Zmoki) [inline rollup.config.js]: I think it can be a simple `uploadcare.js`

File: package.json (hunk 1 above)
Comment (nd0ut) [reply package.json]: @Zmoki yep, `browser` filed is not for IIFE, it is used by bundlers to use as entry point for browsers. It is needed if we have some browser-specific difference between bundles. But we have not them so we do not need `browser` property.

Reference: https://github.com/defunctzombie/package-browser-field-spec

Comment (Zmoki) [reply package.json]: but `module` is the esm bundle, not the source code

Comment (nd0ut) [reply package.json]: @Zmoki `index.js` is not source code. It is entry point that exporting modules from bundles.

```
import * as UploadAPI from './dist/upload.esm'

export {UploadAPI}
```

Also we have `upload.js` which exporting only `UploadAPI`:

```
export * from './dist/upload.esm'
```

**Pull request merged**